soh/src/overlays/actors/ovl_Item_Etcetera/z_item_etcetera.c

```diff
@@ -240,8 +244,16 @@ void ItemEtcetera_Draw(Actor* thisx, GlobalContext* globalCtx) {
     if (gSaveContext.n64ddFlag) {
         if (type == ITEM_ETC_ARROW_FIRE) {
             this->giDrawId = GetItemModelFromId(GetRandomizedItemIdFromKnownCheck(RC_LH_SUN, GI_ARROW_FIRE));
+            s32 randoGetItemId = GetRandomizedItemId(this->getItemId, this->actor.id, this->actor.params, globalCtx->sceneNum);
+            if (randoGetItemId >= GI_MINUET_OF_FOREST && randoGetItemId <= GI_DOUBLE_DEFENSE) {
+                EnItem00_CustomItemsParticles(&this->actor, globalCtx, randoGetItemId);
+            }
         } else if (type == ITEM_ETC_LETTER) {
             this->giDrawId = GetItemModelFromId(GetRandomizedItemIdFromKnownCheck(RC_LH_UNDERWATER_ITEM, GI_LETTER_RUTO));
+            s32 randoGetItemId = GetRandomizedItemId(this->getItemId, this->actor.id, this->actor.params, globalCtx->sceneNum);
```

Comment:
didn't notice this before, but i'm surprised this is working...

it's interesting that it is, but it doesn't make sense for us to do one call to KnownCheck and one using id/params/scene

i'm thinking this whole block can be cleaned up into 
```c
    if (gSaveContext.n64ddFlag) {
        s32 randoGetItemId;
        if (type == ITEM_ETC_ARROW_FIRE) {
            randoGetItemId = GetRandomizedItemIdFromKnownCheck(RC_LH_SUN, GI_ARROW_FIRE);
        } else if (type == ITEM_ETC_LETTER) {
            randoGetItemId = GetRandomizedItemIdFromKnownCheck(RC_LH_UNDERWATER_ITEM, GI_LETTER_RUTO);
        }
        if (randoGetItemId >= GI_MINUET_OF_FOREST && randoGetItemId <= GI_DOUBLE_DEFENSE) {
            EnItem00_CustomItemsParticles(&this->actor, globalCtx, randoGetItemId);
        }
    }
```